src/pages/index.js

```diff
@@ -36,25 +36,85 @@ const profileDescriptionInput = document.querySelector(
   "#profile-decription-input"
 );
 const profileEditForm = profileEditModal.querySelector(".modal__form");
+const profileAvatar = document.querySelector(".profile__image");
 const cardTemplate =
   document.querySelector("#card-template").content.firstElementChild;
 const addNewCardButton = document.querySelector(".profile__add-button");
 
 // Instances of the Classes
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "27d1bd70-108d-4f8b-b35a-2fb4fb8998a7",
+    "Content-Type": "application/json",
+  },
+});
+
+const editAvatarPopup = new PopupWithForm("#avatar-edit-modal", (cardData) => {
+  editAvatarPopup.renderLoading(true, "Saving...");
+
+  api
+    .updateAvatar(cardData)
+    .then((res) => {
+      userInfo.setAvatar(res.avatar);
+
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoading(false);
+    });
+});
+
 const selector = { popupSelector: selectors.previewPopup };
 const cardPreviewPopup = new PopupWithImage(selector);
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api.getUserInfo().then((userData) => {
+  console.log(userData);
+  userInfo.setUserInfo(userData.name, userData.about);
+});
+
+const confirmDeleteModal = new PopupWithConfirmation(
+  "#confirm-delete-modal",
+  (id) => {
+    confirmDeleteModal.renderLoading(true, "Deleting...");
+
+    api
+      .removeCard(id)
+      .then(() => {
+        confirmDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        confirmDeleteModal.renderLoading(false);
```

Comment:
You don't have `renderLoading` in class `PopupWithConfirmation`

Reply:
Hi, I went back to the task instructions and I think I may have misunderstood the instructions. I deleted this from confirmDeleteModal since it didn't ask for that to be done for this modal. Please let me know if I'm incorrect.